crates/astria-auctioneer/src/metrics.rs

```diff
@@ -1,25 +1,153 @@
+            .new_counter_factory(
+                BLOCK_COMMITMENTS_RECEIVED,
+                "the number of block commitments received from the Sequencer node",
+            )?
+            .register()?;
+
+        let executed_blocks_received_count = builder
+            .new_counter_factory(
+                EXECUTED_BLOCKS_RECEIVED,
+                "the number of executed blocks received from the Rollup node",
+            )?
+            .register()?;
+
+        let optimistic_blocks_received_count = builder
+            .new_counter_factory(
+                OPTIMISTIC_BLOCKS_RECEIVED,
+                "the number of optimistic blocks received from the Sequencer node",
+            )?
+            .register()?;
+
+        let auction_bids_received_count = builder
+            .new_counter_factory(
+                AUCTION_BIDS_RECEIVED,
+                "the number of auction bids received from the Rollup node (total number over the \
+                 runtime of auctioneer)",
+            )?
+            .register()?;
+
+        let mut auction_bids_processed_factory = builder.new_gauge_factory(
+            AUCTION_BIDS_PROCESSED,
+            "the number of auction bids processed during an auction (either admitted or dropped \
+             because the time was up or due to some other issue)",
+        )?;
+        let auction_bids_admitted_gauge =
+            auction_bids_processed_factory.register_with_labels(&[(
+                AUCTION_BIDS_PROCESSED_LABEL,
+                AUCTION_BIDS_PROCESSED_ADMITTED.to_string(),
+            )])?;
+        let auction_bids_dropped_gauge =
+            auction_bids_processed_factory.register_with_labels(&[(
+                AUCTION_BIDS_PROCESSED_LABEL,
+                AUCTION_BIDS_PROCESSED_DROPPED.to_string(),
+            )])?;
+
+        let auctions_cancelled_count = builder
+            .new_counter_factory(
+                AUCTIONS_CANCELLED,
+                "the number of auctions that were cancelled due to a proposed block pre-empting a \
+                 prior proposed block",
+            )?
+            .register()?;
+
+        let auctions_submitted_count = builder
+            .new_counter_factory(
+                AUCTIONS_SUBMITTED,
+                "the number of successfully submitted auctions",
+            )?
+            .register()?;
+
+        Ok(Self {
+            auction_bids_admitted_gauge,
+            auction_bids_dropped_gauge,
+            auction_bids_received_count,
+            auctions_submitted_count,
+            auctions_cancelled_count,
+            block_commitments_received_count,
+            executed_blocks_received_count,
+            optimistic_blocks_received_count,
+        })
     }
 }
 
 metric_names!(const METRICS_NAMES:
+    BLOCK_COMMITMENTS_RECEIVED,
+    EXECUTED_BLOCKS_RECEIVED,
+    OPTIMISTIC_BLOCKS_RECEIVED,
+    AUCTIONS_CANCELLED,
+    AUCTIONS_SUBMITTED,
+    AUCTION_BIDS_PROCESSED,
```

Comment:
I am not happy with this name - I want to have a gauge with 2 labels: bids admitted and bids dropped. "processed" seems confusing. Are there better ideas?

Reply:
lets make the bids received and dropped a histogram.

Reply:
See my other comment in https://github.com/astriaorg/astria/pull/1953#discussion_r1946490223, copied here:

> The distinction is important actually (received is processed + dropped), but I did change the nomenclature in 27af6c11 (now the histogram is just "auction_bids" with labels ["processed", "dropped"]).